scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,143 @@
+            stat_var.update(self.features['measurement']['_DEFAULT'])
+
+        # set the populationType attribute
+        if 'populationType' not in stat_var:
+          stat_var['populationType'] = self._get_population_type(part_list)
+
+        # associate pvs to stat_var
+        if 'pvs' in self.features:
+            #p = property, k = dictionary of substrings in column that points to a propertyValue
+            for p, k in self.features['pvs'].items():
+                for c, v in k.items():
+                    ## check if the current key of dict matches with any substring from the tokens of a column name
+                    for part in part_list:
+                        # check if column substring matches with a column token
+                        if c == part:
+                            stat_var[p] = v
+                            # add inferred properties from property 'p' to the stat_var
+                            if 'inferredSpec' in self.features and p in self.features['inferredSpec']:
+                                  stat_var.update(self.features['inferredSpec'][p])
+
+
+      
+        #TODO: If quantityRanges occurs as values in multiple properties, find the property that relates to the column
+        #Suggestion from Prashnath: using _DISAMBIGUATING_POP_TYPE for pvs (income and earnings -- in the test) 
+
+        #TODO: Add units and scalingFactor
+
+        # if the column is Margin of Error, update statType key
+        if 'Margin of Error' in part_list:
+            stat_var['statType'] = 'marginOfError'
+
+        ## add Universe PVs based on the populationType of StatVar
+        dependent_properties = None
+        if 'universePVs' in self.features:
+            for elem in self.features['universePVs']:
+                if stat_var['populationType'] == elem['populationType']:
+                    # check if all constraints of this populationType is in stat_var
+                    if (set(elem['constraintProperties']).issubset(
+                            set(list(stat_var.keys())))):
+                        dependent_properties = list(elem['dependentPVs'].keys())
+
+        # generating dcid using the utils/statvar_dcid_generator.py
+        stat_var['dcid'] = get_stat_var_dcid(stat_var,
+                                             ignore_props=dependent_properties)
+        return stat_var
+
+    def _get_population_type(self, part_list):
+        """From tokenized column name, find the most relevant populationType from the JSON Spec """
+        if 'populationType' in self.features:
+            for part in part_list:
+                for k, v in self.features['populationType'].items():
+                    if k in part:
+                        return self.features['populationType'][k]
+            # If populationType not in column name, take the default from the spec
+            return self.features['populationType']['_DEFAULT']
```

Comment:
self.features['populationType'].get('_DEFAULT') will return None without raising an exception if _DEFAULT doesn;t exist

Reply:
is 'except' executed if there is no 'part' that matches any key in  self.features['populationType']?

Reply:
This is fixed in the [code](https://github.com/datacommonsorg/data/blob/de8483a44a3c0aecfccd217d00a1e5a50a787006/scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py#L392-L405)